tests/functional/mesh/mesh_test.go

```diff
@@ -94,6 +99,56 @@ func TestMeshConnections(t *testing.T) {
 			if connectionsReady == false {
 				t.Error("Timed out while waiting for connections")
 			}
+			mesh.Shutdown()
+		})
+	}
+}
+
+// Test that a mesh starts and that connections are what we expect
+func TestMeshShutdown(t *testing.T) {
+	testTable := []struct {
+		filename string
+	}{
+		{"mesh-definitions/random-mesh-tcp.yaml"},
+		{"mesh-definitions/random-mesh-udp.yaml"},
+		{"mesh-definitions/random-mesh-ws.yaml"},
+	}
+	for _, data := range testTable {
+		filename := data.filename
+		t.Run(filename, func(t *testing.T) {
+			mesh, err := NewMeshFromFile(filename)
+			if err != nil {
+				t.Error(err)
+			}
+			err = mesh.WaitForReady(10000)
+			if err != nil {
+				t.Error(err)
+			}
+			mesh.Shutdown()
+			mesh.WaitForShutdown()
+
+			// Check that the connections are closed
+			pid := os.Getpid()
+			pidString := strconv.Itoa(pid)
+			done := false
+			for timeout := 1000000; timeout > 0 && !done; {
```

Comment:
It's more idiomatic to use the duration constants rather than relying on durations being nanoseconds (and making people count the zeros).  `timeout := 1 * time.Second` would be make it much more immediately apparent what's going on here.